lib/connection/result/column.js

```diff
@@ -550,7 +551,18 @@ function convertRawVariant(rawColumnValue, column, context)
   {
     try
     {
-      ret = eval("(" + rawColumnValue + ")");
+      // check if raw string is in XML format
+      // matches any string with an enclosed '<' and '>' character
+      if ((/^\s*<[\s\S]*>/).test(rawColumnValue))
```

Comment:
Can we do more accurate way to detect XML?

Reply:
It's possible to use XMLValidator to check if the raw string is in valid XML format: https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/v4/4.XMLValidator.md

- XMLValidator returns true if no issue is found.
- XMLValidator returns an error object if any issue is found.

Reply:
My question is more about how to differentiate a string value is an XML or not? Just by checking angle brackets won't be sufficient.

Reply:
PR is updated to have a more accurate way of differentiating between an XML string and a non-XML string
It checks if each tag in the string is closed. It also checks if the attributes and element in each tag are valid

Reply:
The JSON test case is also updated to include angle brackets in the value to test that it's still correctly detected as JSON